### DISCUSSION TITLE
Github GraphQL in NodeJS App: A query attribute must be specified and must be a string.

### DISCUSSION THREAD

**Discussion by madelinezec** [community/community#24814] [Apps, API and Webhooks]
<p>I have two different queries.</p>
<p>The first is in a private repo:</p>
<pre><code>const query = `{
      repository(
        owner: "privateOrg"
        name: "privateRepo"
      ) {
        name
        forkCount
        forks(
          first: 27
          orderBy: { field: NAME, direction: DESC }
        ) {
          totalCount
          nodes {
            name
          }
        }
      }
    }`
</code></pre>
<p>The second is in a public repo:</p>
<pre><code>const querytwo = `{
      repository(
        owner: "publicOrg"
        name: "publicRepo"
      ) {
        name
        forkCount
        forks(
          first: 11
          orderBy: { field: NAME, direction: DESC }
        ) {
          totalCount
          nodes {
            name
          }
        }
      }
    }`
</code></pre>
<p>The fetch for both look identical except the queries:</p>
<pre><code>fetch('https://api.github.com/graphql', {
  method: 'POST',
  body: JSON.stringify({query}),
  headers: {
    'Authorization': `Bearer ${accessToken}`,
  },
}).then(res =&gt; res.text())
  .then(body =&gt; console.log(body))
  .catch(error =&gt; console.error(error));
  console.log("\n\n\n");

    fetch('https://api.github.com/graphql', {
      method: 'POST',
      body: JSON.stringify({querytwo}),
      headers: {
        'Authorization': `Bearer ${accessToken}`,
      },
    }).then(res =&gt; res.text())
      .then(body =&gt; console.log(body))
      .catch(error =&gt; console.error(error));
    console.log("\n\n\n");
</code></pre>
<p>The first query returns: </p>
<pre><code>{"data":{"repository":{"name":"mms-docs","forkCount":26,"forks":{"totalCount":8,"nodes":[{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"},{"name":"mms-docs"}]}}}}
</code></pre>
<p>The second query returns:</p>
<pre><code>{"errors":[{"message":"A query attribute must be specified and must be a string."}]}
</code></pre>
<p>What am I doing wrong? </p>

**Comment by madelinezec** [comment, marked as answer]
<p>The issue was short hand object notation. </p>
<pre><code>JSON.stringify({query})
</code></pre>
<p>is shorthand for </p>
<pre><code>JSON.stringify({query: query})
</code></pre>
<p>which becomes</p>
<pre><code>{
query:
 {
      repository(
        owner: "PrivateOrg"
        name: "privateRepoName"
      ) {
        name
        forkCount
        forks(
          first: 11
          orderBy: { field: NAME, direction: DESC }
        ) {
          totalCount
          nodes {
            name
          }
        }
      }
    }
   }`
</code></pre>
<p><strong>JSON.stringify({querytwo})</strong> is shorthand for <strong>JSON.stringify({querytwo: querytwo})</strong>, or in other words it creates this query:</p>
<pre><code>{
querytwo:
 {
      repository(
        owner: "PrivateOrg"
        name: "privateRepoName"
      ) {
        name
        forkCount
        forks(
          first: 11
          orderBy: { field: NAME, direction: DESC }
        ) {
          totalCount
          nodes {
            name
          }
        }
      }
    }
   }`
</code></pre>
<p>Hence why GraphQL couldn’t find a  <strong>query</strong>  - it found a  <strong>queryTwo</strong></p>

**Comment by pro-master** [comment]
<p>I need help I’m new I’m kind of confuse</p>
